src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Assignment.cs

```diff
@@ -14,10 +16,12 @@ public ExprAssignment(Expr lhs, Expr rhs)
             Flags = EXPRFLAG.EXF_ASSGOP;
         }
 
-        public Expr LHS { get; set; }
+        public Expr LHS
+        {
+            get => _lhs;
+            set => Type = (_lhs = value).Type;
```

Comment:
It can and it will. Doing so hits enough points I know will conflict with open PRs that I'm waiting until I've fewer open.